app/controllers/concerns/two_factor_authenticatable.rb

```diff
@@ -160,6 +160,23 @@ def update_phone_attributes
         user: current_user,
         attributes: { phone: user_session[:unconfirmed_phone], phone_confirmed_at: Time.zone.now }
       ).call
+      # Eventually, this will be brought into this method once we only write to the
+      # phone_configurations table
+      update_phone_configuration
+    end
+  end
+
+  def update_phone_configuration
+    # Check if the user has a phone_configuration -- if not, then create one
+    # for now, the user has at most one phone configuration
+    attributes = { phone: user_session[:unconfirmed_phone],
+                   confirmed_at: Time.zone.now,
+                   delivery_preference: two_factor_authentication_method }
+    phone_configuration = current_user.phone_configuration
+    if phone_configuration.present?
+      phone_configuration.update!(attributes)
+    else
+      current_user.create_phone_configuration(attributes)
```

Comment:
What about using `find_or_create_by`?

Reply:
That would work as well.

Reply:
I ended up moving this logic to the `UpdateUser` service object. What we really need is a `create_or_update` method, which doesn't seem to be part of ActiveRecord from what I can find.